R/metrics.R

```diff
@@ -179,8 +178,8 @@ metrics.data.frame <- function(data,
 #' predictions (the class probability columns) as bare column names or
 #' `tidyselect` selectors to `...`.
 #'
-#' When mixing dynamic and static survival metrics, `censoring_weights` and
-#' `eval_time` is only needed when dynamic metrics are included in the set.
+#' When mixing dynamic and static survival metrics, `pred_time` is used to
```

Comment:
I'd like to change this to be similar to how it works with classification: 

```r
lung_surv |>
  concordance_survival(truth = surv_obj, estimate = .pred_time)

lung_surv |>
  brier_survival(truth = surv_obj,  .pred_survival)

lung_surv |>
  mixed_set(truth = surv_obj, estimate = .pred_time,  .pred_survival)
```

IMO this is most analogous to the classification scenario where `estimate` is the point estimate (hard class prediction / time prediction) and ... maps to the probability estimates (class probabilities / survival prediction).  Not a perfect analogy due the static and dynamic split but I think that it works better.

Reply:
Done!

``` r
library(yardstick)

lung_surv |>
  concordance_survival(truth = surv_obj, estimate = .pred_time)
#> # A tibble: 1 × 3
#>   .metric              .estimator .estimate
#>   <chr>                <chr>          <dbl>
#> 1 concordance_survival standard       0.637

lung_surv |>
  brier_survival(truth = surv_obj,  .pred)
#> # A tibble: 5 × 4
#>   .metric        .estimator .eval_time .estimate
#>   <chr>          <chr>           <dbl>     <dbl>
#> 1 brier_survival standard          100     0.109
#> 2 brier_survival standard          200     0.194
#> 3 brier_survival standard          300     0.219
#> 4 brier_survival standard          400     0.222
#> 5 brier_survival standard          500     0.197

mixed_set <- metric_set(concordance_survival, brier_survival)

lung_surv |>
  mixed_set(truth = surv_obj, estimate = .pred_time,  .pred)
#> # A tibble: 6 × 4
#>   .metric              .estimator .eval_time .estimate
#>   <chr>                <chr>           <dbl>     <dbl>
#> 1 brier_survival       standard          100     0.109
#> 2 brier_survival       standard          200     0.194
#> 3 brier_survival       standard          300     0.219
#> 4 brier_survival       standard          400     0.222
#> 5 brier_survival       standard          500     0.197
#> 6 concordance_survival standard           NA     0.637
```

<sup>Created on 2023-04-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>